### PR TITLE
Add documentation for R support.

### PR DESCRIPTION
**Pull request by craigcitro**
This is the documentation complement to [this travis-build pull
request](https://github.com/travis-ci/travis-build/pull/386).

### CODE REVIEW

**Comment by craigcitro** [comment]
Matches travis-ci/travis-build#386.

PTAL @eddelbuettel and @hadley, and feel free to have some others rip into the docs.

File: user/languages/r.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+Travis CI support for R is contributed by the community and may be removed or
+altered at any time. If you run into any problems, please report them in the
+[Travis CI issue tracker](https://github.com/travis-ci/travis-ci/issues/new?labels=community:r)
+and cc [@craigcitro](https://github.com/craigcitro),
+[@eddelbuettel](https://github.com/eddelbuettel), and
+[@hadley](https://github.com/hadley).
+
+## Basic configuration
+
+If your R repository doesn't need any dependencies beyond those specified in
```
Comment (hadley) [inline user/languages/r.md]: Maybe start with: "R support is designed for testing packages."?

File: user/languages/r.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+## Configuration options
+
+Travis CI supports a number of configuration options for your R package.
+
+### Dependencies
+
+By default, Travis CI will install all R packages listed as dependencies in
+your package `DESCRIPTION` file. However, it's often necessary or desirable to
+install additional dependencies, or newer versions of those dependencies.
```
Comment (hadley) [inline user/languages/r.md]: (or to install binary packages to decrease install time)

File: user/languages/r.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  become failures. This is especially helpful when preparing your package for
+  submission to CRAN, and is recommended for most packages.
+
+* `r_build_args`: additional arguments to pass to `R CMD build`, as a single
+  string. Defaults to empty.
+
+* `r_check_args`: additional arguments to pass to `R CMD check`, as a single
+  string. Defaults to `--as-cran`.
+
+* `r_check_cran_incoming`: if `true`, sets the value of the
```
Comment (hadley) [inline user/languages/r.md]: If the default is now to use `R CMD check` and not `devtools::check()`, then this isn't needed

File: user/languages/r.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+A simple BioConductor package should generally be able to use Travis CI with
+the following `.travis.yml`:
+
+    language: r
+    bioc_required: true
+
+### Miscellaneous
+
+* `pandoc`: If `true`, a recent version of
```
Comment (hadley) [inline user/languages/r.md]: Can we make this the default?

File: user/languages/r.md (hunk 1 above)
Comment (craigcitro) [inline user/languages/r.md]: Added an introductory sentence, but I don't know if I should say more. (I'm tempted to also link to your book ...)

File: user/languages/r.md (hunk 2 above)
Comment (craigcitro) [inline user/languages/r.md]: i actually reworked this section a little; better? worse?

File: user/languages/r.md (hunk 3 above)
Comment (craigcitro) [inline user/languages/r.md]: isn't this still needed to pick up some checks? (i'm basing this solely on the [r internals](http://cran.r-project.org/doc/manuals/R-ints.html) manual, not any actual information.)

File: user/languages/r.md (hunk 4 above)
Comment (craigcitro) [inline user/languages/r.md]: good call. done.

**Comment by BanzaiMan** [comment]
It must be noted that R requires `sudo` to run, so that it cannot be used in the containers.

File: user/languages/r.md (hunk 3 above)
Comment (hadley) [inline user/languages/r.md]: No, devtools / `--as-cran` takes care of that for you.

**Comment by hadley** [comment]
@BanzaiMan you mean the current setup script requires `sudo`, right? (because R certainly doesn't)

**Comment by eddelbuettel** [comment]
@Banzaiman There is no `sudo R` in the script; `sudo` is needed chiefly for package management at the `deb` level.

**Comment by BanzaiMan** [comment]
I think there is at least one invocation of `sudo` inside `install`, which I believe occurs after `sudo` is disabled. https://github.com/travis-ci/travis-build/pull/386/files#diff-5b7e1aa01f1c6f058df6e0b9e2486c62R229 Please try it with `sudo: false`.

**Comment by eddelbuettel** [comment]
That is just what I claimed it was:  `sudo apt-get install`.  

Now this is @craigcitro 's code and not mine, but there is simply no way to invoke `apt-get install` without being root.   The only way to have `sudo` removed is to ensure the segment is already running as root but I cannot speak to that...

**Comment by hadley** [comment]
@BanzaiMan The `apt-get install` option is a performance optimisation - many R packages are available as pre-built binaries, so obviously installing those is much faster than compiling from scratch. What are the consequences of requiring sudo? What are the consequences of this script not being available in the containers?

**Comment by BanzaiMan** [comment]
The use of `sudo` should be discussed in the PR. Sorry about the diversion.

File: user/languages/r.md (hunk 3 above)
Comment (craigcitro) [inline user/languages/r.md]: sweet! done. (here and in the travis code.)

**Comment by craigcitro** [comment]
OK, updated with a containers comment.

(Let me know if you want me to flesh it out more, or make it clear under what conditions we can/can't use containers. As mentioned in the other PR, I'm ultimately hoping we can remove this restriction in the future.)

**Comment by BanzaiMan** [comment]
@craigcitro It just occur to me; should there be a sort of "migration guide" for users who's using r-travis already?

**Comment by eddelbuettel** [comment]
Yes there should be and I am pretty sure there will be ...

**Comment by craigcitro** [comment]
I just wrote the start of one last night, and stuck it on the r-travis wiki:
  https://github.com/craigcitro/r-travis/wiki/Porting-to-native-R-support-in-Travis

there's a link here, but let me feature that a bit more prominently ...

**Comment by craigcitro** [comment]
@eddelbuettel @BanzaiMan better?

**Comment by eddelbuettel** [comment]
Not bad!  

How would I add personal PPAs now that I started (again) to populate [ppa:edd/misc](https://launchpad.net/~edd/+archive/ubuntu/misc/+packages) ? Standard Travis script lines ie

```
before_install:                                                                                                                                                                                                                           
  - sudo add-apt-repository -y ppa:edd/misc
  - sudo apt-get update -q
```

Also, could you clarify in the wiki which goodies are, or are not, part of the (expanded ?) base image?

**Comment by BanzaiMan** [comment]
@craigcitro Great. I'll add a link to the guide from the announcement post as well.

**Pull request merged**